tensorboard/plugins/beholder/beholder_plugin.py

```diff
@@ -96,18 +98,14 @@ def _serve_is_active(self, request):
     }
     return http_util.Respond(request, response, 'application/json')
 
-
   def _fetch_current_frame(self):
     path = '{}/{}'.format(self.PLUGIN_LOGDIR, shared_config.SUMMARY_FILENAME)
-
-    try:
-      frame = file_system_tools.read_tensor_summary(path).astype(np.uint8)
-      self.most_recent_frame = frame
-      return frame
-
-    except (message.DecodeError, IOError, tf.errors.NotFoundError):
-      return self.most_recent_frame
-
+    with self._lock:  # only to put boundaries on workload for now
```

Comment:
I don't understand this comment - what does the lock accomplish?  This code should be called synchronously from the `_frame_generator()` helper so the locking only would come into play if we were somehow streaming via two requests at once, which shouldn't normally be possible in the front end logic.  Also, if we did want to bound workload during multiple streams, it would make more sense to have the current frame fetching logic occurring in only a single thread.  This change would instead result in two request threads competing to fetch frames, so you might end up sort of round-robining frames to the two streams which seems very weird.

Reply:
`_frame_generator()` is an endpoint for a multi-threaded HTTP server. I agree that a single thread should be what's computing this for all viewers. Might not be worth the effort quite yet, since it's likely there'll only be a single viewer. We don't have the resources quite yet to make Beholder scale. But we will in the future.